gtsam/hybrid/HybridFactorGraph.cpp

```diff
@@ -58,13 +57,49 @@ const gtsam::DiscreteFactorGraph& HybridFactorGraph::discreteGraph() const {
   return discreteGraph_;
 }
 
+const GaussianFactorGraph& HybridFactorGraph::gaussianGraph() const {
+  return gaussianGraph_;
+}
+
+HybridFactorGraph HybridFactorGraph::linearize(
+    const Values& continuousValues) const {
+  // linearize the continuous factors
+  auto gaussian_factor_graph = nonlinearGraph_.linearize(continuousValues);
+
+  // linearize the DCFactors
+  DCFactorGraph linearized_DC_factors;
+  for (DCFactor::shared_ptr factor : dcGraph_) {
+    // Check if the factor is a DCGaussianMixtureFactor since we don't linearize
+    // those.
+    auto dcgm = dynamic_cast<const DCGaussianMixtureFactor*>(factor.get());
```

Comment:
I think you can do
`auto dcgm = dynamic_pointer_cast<const DCGaussianMixtureFactor>(factor);`

Reply:
Gotcha. I was trying to avoid an extra include.

Reply:
Hmm doesn't work. I guess this is because `factor` is a `boost::shared_ptr`?

Reply:
Yeah that is it. I see this in the error message: `note:   ‘gtsam::DCFactor’ is not derived from ‘const std::shared_ptr<_Tp>’`